ofdb-gateways/src/mailgun.rs

```diff
@@ -0,0 +1,66 @@
+use ofdb_entities::{event::*, place::*};
+
+// TODO: move into ofdb-core crate
+/// The kind of notification.
+#[derive(Debug, PartialEq)]
+pub enum Notification {
+    PlaceAdded(Place),
+    PlaceUpdated(Place),
+    EventAdded(Event),
+    EventUpdated(Event),
+}
+
+// TODO: move into ofdb-core crate
+/// A subsystem that manages everything concerning
+/// notifications.
+pub trait NotificationManager {
+    fn notify(&self, n: Notification);
+}
+
+/// A email notification manager.
+pub struct Manager {
+    pub api_key: String,
+    pub domain: String,
+    pub from_email: String,
+    pub to_email: String,
+}
+
+impl NotificationManager for Manager {
+    fn notify(&self, n: Notification) {
```

Comment:
TODO: Send notifications from a separate thread and return immediately from the trait method.